File: GamesTournamentsWeb.Web/ClientApp/models/user/HistoryItem.ts
import { ConvertableToJson } from '~/models/ConvertableToJson'

class HistoryItem extends ConvertableToJson {
  public accountId: number
  public gameId: number
  public gameName: string
  public place: number
  public tournamentId: number

  constructor (accountId: number, gameId: number, gameName: string, place: number, tournamentId: number) {
    super()

    this.accountId = accountId
    this.gameId = gameId
    this.gameName = gameName
    this.place = place
    this.tournamentId = tournamentId
  }
}

export { HistoryItem }

File: GamesTournamentsWeb.Infrastructure/Dto/Users/HistoryItem.cs
namespace GamesTournamentsWeb.Infrastructure.Dto.Users;

public class HistoryItem
{
    public int AccountId { get; set; }
    public int GameId { get; set; }
    public string GameName { get; set; }
    public string GameUsername { get; set; }
    public int TournamentId { get; set; }
    public string TournamentName { get; set; }
}